### ISSUE TITLE
glibc problems with pre-built node-mapnik 3.5.14

### ISSUE THREAD

**Issue by pnorman** [mapnik/node-mapnik#700]
```
pnorman@pippin:~/osm/kosmtik-3$ ./index.js --help

module.js:356
  Module._extensions[extension](this, filename);
                               ^
Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/pnorman/osm/kosmtik-3/node_modules/mapnik/lib/binding/node-v11-linux-x64/map
nik.node)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/pnorman/osm/kosmtik-3/node_modules/mapnik/lib/mapnik.js:31:15)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

Changing the mapnik version 3.5.13 fixed the error.

If this is a new requirement in Mapnik, then with semver the node-mapnik versionn have a bump of more than just patch version.

cross-ref https://github.com/kosmtik/kosmtik/pull/168

**Comment by nyurik** [comment]
We just ran into the same issue for Kartotherian, and had to use 3.5.13 via shrinkwrap.

**Comment by tjwebb** [comment]
Same here. We've been stuck on 3.5.13 for awhile now: https://github.com/langateam/node-mapnik

**Comment by springmeyer** [comment]
This can be solved per the README (https://github.com/mapnik/node-mapnik#depends):

```
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update -y
sudo apt-get install -y libstdc++-5-dev
```

This has always been required to run node-mapnik binaries on ubuntu precise. The change is that this is now required on ubuntu trusty as well. This is not required on ubuntu Xenial (since the libstdc++ there is recent enough to have the required GLIBCXX).

Yes, this requirement (needed to start upgrading libstdc++ on trusty whereas before you were not required) is new and ideally should have been in a minor or major release. I'm working on trying to find a way to release a 3.5.15 with a slightly lower `GLIBCXX` requirement, but it is not an easy problem to solve and I make no promises I can provide that. So, upgrading libstdc++, per the readme is the best approach rather than waiting.

**Comment by nyurik** [comment]
@springmeyer I'm not sure Debian Jessie has support for that version yet, and all of our servers are running it. For now sadly our best option is to revert to 3.5.13, hoping for the improved patch.

**Comment by tjwebb** [comment]
@springmeyer the main problem is that you broke semver by releasing 3.5.14 as a patch, so everyone who `npm install -s` and has `^3.5.0` in their package.json, subsequent `npm install` and deployments now pull down 3.5.14, which contains all these new requirements. According to semver, breaking changes constitute a major-version release.

It seems that libmapnik does not follow semver, which is fine, but [you need to follow semver](https://docs.npmjs.com/getting-started/semantic-versioning#semver-for-publishers) if you're publishing to `npm`.

**Comment by brianreavis** [comment]
@tjwebb See https://github.com/mapnik/node-mapnik/pull/698#issuecomment-247097148

**Comment by springmeyer** [comment]
> I'm working on trying to find a way to release a 3.5.15 with a slightly lower GLIBCXX requirement, but it is not an easy problem to solve and I make no promises I can provide that. So, upgrading libstdc++, per the readme is the best approach rather than waiting.

This will be resolved by the upcoming node-mapnik v3.6.3 release: https://github.com/mapnik/node-mapnik/pull/839

**Comment by pnorman** [comment]
It leaves 3.5.14 as the last 3.5 release that follows semver :(
